android/app/src/androidTest/java/com/bc/ur/URDecoderTest.kt

```diff
@@ -23,26 +23,48 @@ class URDecoderTest {
     @Test
     fun testDecodeMultiParts() {
         val ur = UR.create(32767, "Wolf")
+
         val encoder = UREncoder(ur, 1000, 100, 10)
         val decoder = URDecoder()
-        do {
-            val part = encoder.nextPart()
-            decoder.receivePart(part)
-        } while (!decoder.isComplete)
 
-        // make sure resultUR return exact UR entered before
-        val resultUR = decoder.resultUR()
-        assertEquals(ur.type, resultUR.type)
-        assertTrue(
-            ur.cbor.toTypedArray().contentDeepEquals(resultUR.cbor.toTypedArray())
-        )
+        encoder.use { e ->
+            decoder.use { d ->
+                do {
+                    val part = e.nextPart()
+                    d.receivePart(part)
+                } while (!d.isComplete)
+
+                // make sure resultUR return exact UR entered before
+                val resultUR = d.resultUR()
+                assertEquals(ur.type, resultUR.type)
+                assertTrue(
+                    ur.cbor.toTypedArray().contentDeepEquals(resultUR.cbor.toTypedArray())
+                )
+
+                // make sure getting resultError throw IllegalStateException
+                try {
+                    d.resultError()
```

Comment:
`assertFailsWith` would be better here

Reply:
@devrandom I'm using JUnit4 and it seems like `assertFailsWith` is only added from JUnit5. Never mind, i can simplify that by define by myself.